blackbird_python/blackbird/listener.py

```diff
@@ -447,26 +445,39 @@ def exitStatement(self, ctx: blackbirdParser.StatementContext):
             self._program._operations.append(operation)
 
     def enterForloop(self, ctx: blackbirdParser.ForloopContext):
-        self._infor = True
+        self._in_for = True
 
     def exitForloop(self, ctx: blackbirdParser.ForloopContext):
-        self._infor = False
+        self._in_for = False
 
-        if ctx.INT():
-            _FORVAR = range(int(ctx.INT().getText()))
+        if ctx.rangeval():
```

Comment:
Just double checking, but the range syntax can only be used inside a for loop. If it is used elsewhere, it will either (a) raise a syntax error (because the lexer won't be expecting it), or (b) if it passes the lexer, will be ignored by the parser since `if ctx.rangeval()` isn't included anywhere else.

Reply:
That's correct! It would be possible to add it as an allowed expression and thus be able to store it to an array, but I'm not sure that's worth it at the moment. I can't think of a clear use-case for it really. :thinking: